### ISSUE TITLE
fails to find libgit2

### ISSUE THREAD

**Issue by raimondi1337** [libgit2/pygit2#603]
System: Arch Linux
libgit2 version: 0.24.0
python version: 3.5.1
pip version: 8.1.0

not fixed by:
$ sudo ldconfig

output:
$python -c 'import pygit2'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.5/site-packages/pygit2/**init**.py", line 32, in <module>
    from _pygit2 import *
ImportError: libgit2.so.24: cannot open shared object file: No such file or directory

**Comment by pypingou** [comment]
Which version of pygit2 are you using?

**Comment by raimondi1337** [comment]
0.24.0

**Comment by jdavid** [comment]
Try this command:

  $ ldd /usr/lib/python3.5/site-packages/_pygit2.so

What it shows?

**Comment by raimondi1337** [comment]
ldd: /usr/lib/python3.5/site-packages/_pygit2.so: No such file or directory

That's strange, first time I tried 'pip install pygit2' it installed, but this time it gives:

Requirement already satisfied (use --upgrade to upgrade): pygit2 in /usr/lib/python3.5/site-packages
Requirement already satisfied (use --upgrade to upgrade): cffi in /usr/lib/python3.5/site-packages (from pygit2)
Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python3.5/site-packages (from pygit2)
Requirement already satisfied (use --upgrade to upgrade): pycparser in /usr/lib/python3.5/site-packages (from cffi->pygit2)

That's probably my issue

**Comment by arjamizo** [comment]
@raimondi1337: did you manage to solve it?

**Comment by raimondi1337** [comment]
nope

**Comment by GeorgeL9** [comment]
I am having the same issue.

```
$pip install pygit2
Requirement already satisfied (use --upgrade to upgrade): pygit2 in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): cffi in /usr/local/lib/python2.7/dist-packages (from pygit2)
Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python2.7/dist-packages (from pygit2)
Requirement already satisfied (use --upgrade to upgrade): pycparser in /usr/local/lib/python2.7/dist-packages (from cffi->pygit2)
```
```
 File "fetcher.py", line 3, in <module>
    import pygit2
  File "/home/customer/venv/local/lib/python2.7/site-packages/pygit2/__init__.py", line 32, in <module>
    from _pygit2 import *
ImportError: libgit2.so.24: cannot open shared object file: No such file or directory
```

**Comment by GeorgeL9** [comment]
Solved!
Per the docs:

```$ sudo ldconfig
$ python -c 'import pygit2'```

**Comment by xiaoanyunfei** [comment]
My OS is CentOS release 6.5, and I met the same problem.
The reason of the problem maybe that **libgit2.so.0.24.0** was put into the wrong directory(/usr/local/lib)
But **ldconfig**default to detect /lilb and /usr/lib
if you execute `sudo ldconfig -v | grep git` ,and don't see libgit2.so.0.24.0,you can fix as below

so we need copy /usr/local/lib/libgit2.so.0.24.0 to /usr/lib64, and rebuild the soft link
`#cp  /usr/local/lib/libgit2.so.0.24.0  /usr/lib64`
`cd /usr/lib64`
`sudo ln -s libgit2.so.0.24.0 libgit2.so.24`
`sudo ln -s libgit2.so.24 libgit2.so`

reexecute `sudo ldconfig -v | grep git` 
       libgit2.so.24 (libc6,x86-64) => /usr/lib64/libgit2.so.24
        libgit2.so (libc6,x86-64) => /usr/lib64/libgit2.so
Then you can use python -c "import pygit2"

**Comment by xiaoanyunfei** [comment]
My os is CentOS release 6.5 (Final),and I have the same problem
The reason maybe the **libgit2.so.0.24.0** was put into wrong diretcory(/usr/local/lib)
But command **ldconfig** default to detect /lilb and /usr/lib
execute `# ldconfig -p | grep libgit2`,if **libgit2.so.0.24.0**  not in output, you can do 
as below
`#cp /usr/local/lib/libgit2.so.0.24.0 /usr/lib`
`#cd /usr/lib`
`#ln -s libgit2.so.0.24.0 libgit2.so.24`
`#ln -s libgit2.so.24 libgit2.so`
if your output is same as below ,the problem maybe is fixed
`# ldconfig -p | grep libgit2`
        libgit2.so.24 (libc6,x86-64) => /usr/lib64/libgit2.so.24
        libgit2.so (libc6,x86-64) => /usr/lib64/libgit2.so

**Comment by xiaoanyunfei** [comment]
@GeorgeL9 
your method can fix the problem on ubuntu 14.04 ,but not on centos6.5

**Comment by raywonteja93** [comment]
Hi,
I am installing libgit2 and pygit2 manually in CentOS 7.
I could install both of them but I have a problem.

![ldd](https://cloud.githubusercontent.com/assets/13003675/23256432/bdc5484c-f9bf-11e6-834c-17a4594a46ae.PNG)
![locate](https://cloud.githubusercontent.com/assets/13003675/23256431/bdc36fea-f9bf-11e6-9593-88afe7459893.PNG)

Can someone help me fix this.
Thanks

**Comment by ochanje210** [comment]
I had this issue on AWS ElasticBeanstalk Python platform with 64bit Amazon Linux 2017.09 v2.6.0

I was able to solve this with the following:
```bash
$ ldconfig /usr/local/lib/
```

**Comment by suhaibchishti** [comment]
On EC2, this would be the way to fix it.

$ ldconfig /usr/local/lib/

**Comment by webknjaz** [comment]
@raimondi1337 this should now be addressed by `0.28.0` since it ships pre-build manylinux1 wheels.

@jdavid this should probably be closed now.
